lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -211,6 +211,29 @@ def set_list_position(new_position, raise_exception_if_save_fails=false)
           raise_exception_if_save_fails ? save! : save
         end
 
+        # Checks if a list has no repeating indices 
+        # Returns true if at least one value position_column has a duplicate value
+        def does_list_have_duplicates?
+          acts_as_list_class.group(position_column).having("count(?) > 1", position_column).any?
+        end
+
+        # Repairs a list that has duplicate positions by moving each of the duplicated up in position until there are not more duplicates
+        # NOTE: We cannot ensure a repeatable order of the duplicate entries since their position infomration was lost upon duplication
+        def repair_duplicate_positions_in_list
+          # While there are duplicates we will increment the position of the second repetition for every number
+          while does_list_have_duplicates? do
+            # Get the position with repetitions
+            repeated_pos = acts_as_list_class.group(position_column).having("count(?) > 1", position_column).first.send(position_column)
+            # Get one of the entries in the given position. This will be the entry that will keep the current position
+            first_repeated_entry_id = acts_as_list_class.where("#{position_column} >= ?", repeated_pos).first.id
```

Comment:
Shouldn't this be `=` not `>=`? And if so, you can simplify the where call to remove the string interpolation :)